### PR TITLE
make sure interface up successful

### LINKED ISSUE #1247: docker run failed on overlay network since "failed to set link up: transport endpoint is not connected"
### docker version:  
1.11.2  on ubuntu 14.04  

### docker network ls:
```
NETWORK ID          NAME                 DRIVER
6a1fa16db582        bridge               bridge
0aa8655e6fbc        host                 host
35873009d789        multi-host-network   overlay
f9705d542f03        none                 null
```

### uname -a:
```
Linux c55c4e23c9d714361aa1be6b7a97e7c63-node1 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

When I start container on overlay network, docker daemon return error：
```
root@c55c4e23c9d714361aa1be6b7a97e7c63-node1:~# docker run -it --net multi-host-network busybox
docker: Error response from daemon: subnet sandbox join failed for "172.19.0.0/16": vxlan interface creation failed for subnet "172.19.0.0/16": failed to set link up: transport endpoint is not connected.
```
And I still can't start the container after restart docker daemon service.
I produce this error many times at 1.11.x, but I don't know how to reproduce this, and how can I workaround this problem without restart host.

### PR DESCRIPTION
**Pull request by BSWANG**
add three times retry to make sure interface up successful
fix #1247

### CODE REVIEW

File: osl/interface_linux.go [hunk 1]
```diff
@@ -290,7 +291,17 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
 	}
 
 	// Up the interface.
-	if err := nlh.LinkSetUp(iface); err != nil {
+	maxSetupRetry := 3
+	retryCount := 0
+	for ; retryCount < maxSetupRetry; retryCount++ {
```
Comment (aboch) [inline osl/interface_linux.go]: Thanks for the change.
I think the following code can be simpified by:
```
	cnt := 0
	for err = nlh.LinkSetUp(iface); err != nil && icnt < 3; cnt++ {
		log.Debugf("retrying link setup after: %v", err)
		time.Sleep(1 * time.Second)
	}
	if err != nil {
		return fmt.Errorf("failed to set link up: %v", err)
	}
```

File: osl/interface_linux.go [hunk 2]
```diff
@@ -290,7 +291,17 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
...
+		if err := nlh.LinkSetUp(iface); err == nil {
+			break
+		} else {
+			log.Debugf("error in setup link up, error %+v", err)
+		}
+		time.Sleep(1 * time.Second)
```
Comment (aboch) [inline osl/interface_linux.go]: Based on your experiment, do we really need to wait a full second. If you have a consistent way to reproduce the issue, can you try smaller wait time, in the order of tens of milliseconds ?
Thanks

Comment (BSWANG) [inline osl/interface_linux.go]: yeah, tens of milliseconds works for me too, I commit it agent as your suggestion.

**Comment by aboch** [comment]
LGTM

**Pull request merged**